### ISSUE TITLE
Compilation fails with gcc version 13 and up

### ISSUE THREAD

**Issue by OlafurSiemsen** [mumax/3#342]
### Steps to reproduce:
Run `make realclean && make` with `gcc` pointing to gcc of version 13+
### Expected behavior:
Compilation
### Actual behavior:
Compilation fails, the last output along with error message being
```powershell
>go build cuda2go.go
nvcc -std=c++03 -ccbin=/usr/bin/gcc --compiler-options -Werror --compiler-options -Wall -Xptxas -O3 -ptx -arch=compute_50 -code=sm_50 copypadmul2.cu -o copypadmul2_50.ptx
In file included from /usr/include/cuda_runtime.h:83,
                 from <command-line>:
/usr/include/crt/host_config.h:132:2: error: #error -- unsupported GNU version! gcc versions later than 12 are not supported! The nvcc flag '-allow-unsupported-compiler' can be used to override this version check; however, using an unsupported host compiler may cause compilation failure or incorrect run time execution. Use at your own risk.
  132 | #error -- unsupported GNU version! gcc versions later than 12 are not supported! The nvcc flag '-allow-unsupported-compiler' can be used to override this version check; however, using an unsupported host compiler may cause compilation failure or incorrect run time execution. Use at your own risk.
make[1]: *** [Makefile:72: copypadmul2_wrapper.go] Error 1
make[1]: Leaving directory '/home/olafur/go/src/github.com/mumax/3/cuda'
make: *** [Makefile:18: cudakernels] Error 2
``` 

### Workaround
One may simply edit the line in the make file to point at an earlier version of `gcc`  one has installed.
https://github.com/mumax/3/blob/f91389b2dcd7e8c9ab17a47d1fa098750215ad1a/cuda/Makefile#L42
Modifying this to `NVCC_CCBIN=/usr/bin/gcc-12`, for example.

### Suggested solution
Add some logic that looks at the user's system and picks an appropriate version of `gcc`.

**Comment by JonathanMaes** [comment]
[This StackOverflow answer](https://stackoverflow.com/a/46380601) lists the highest supported `gcc` version for each CUDA version. I tried adding logic to select the highest supported and installed `gcc` automatically, but this made the Makefile extremely convoluted, so I refrained from making this change.

Instead, **the main Makefile now supports the `NVCC_CCBIN` argument** (e.g. `make NVCC_CCBIN="/usr/bin/gcc-11"`), **which can be used to control the GCC version used**. Alternatively, setting the `NVCC_CCBIN` environment variable should achieve the same result in a more permanent way. I added this information to the new README.

The `deploy/deploy_linux.bash` script has not been changed (apart from an additional warning comment), but this script is normally only used by the core mumax³ team.

I am open to alternate suggestions, though I would refrain from hardcoding a lookup table with GCC/CUDA version compatibility as this would have to be maintained manually.
